R/conda.R

```diff
@@ -758,17 +758,21 @@ conda_run <- function(cmd, args = c(), conda = "auto", envname = NULL,
                       run_args = c("--no-capture-output"), ...) {
 
   conda <- conda_binary(conda)
-  envname <- condaenv_resolve(envname)
+  if(!identical(envname, "base"))
+    envname <- condaenv_resolve(envname)
 
   if (numeric_conda_version(conda) < "4.9")
     stopf(
 "`conda_run()` requires conda version >= 4.9.
 Run `miniconda_update('%s')` to update conda.", conda)
 
-  if(grepl("[/\\]", envname))
+  if(envname == "base") 
+    in_env <- ""
```

Comment:
Can we still set a prefix here explicitly? My worry is that the user might have the `CONDA_PREFIX` environment variable defined (https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#environment-variables-that-affect-the-build-process) and that could override an attempt to use the base environment.

Alternatively, we might need to unset that environment variable before running Conda. What do you think?

Reply:
Thank you for your comments. I hadn't realized that you can explicitly request the base env by name.

I removed all the special handling for `envname = "base"` in `conda_run()`. 
`condaenv_resolve()` already does the right thing where `condaenv_resolve("base") == "base"`, and `/path/to/conda run --name base ...` works also. 

I also confirmed that the command line argument `--name base` takes precedence over the env var `CONDA_PREFIX`.